README.md

```diff
@@ -332,6 +332,20 @@ public function test_special_function() {
 
 It's important to note that the `$priority` and `$parameter_count` arguments (parameters 3 and 4 for both `add_action()` and `add_filter()`) are significant. If `special_function()` were to call `add_action( 'save_post', 'special_save_post', 99, 3 )` instead of the expected `add_action( 'save_post', 'special_save_post', 10, 2 )`, our test would fail.
 
+#### Asserting that Closure hooks have been added
+
+Sometimes it's handy to add a [Closure](https://secure.php.net/manual/en/class.closure.php) as a WordPress hook instead of defining a function in the global namespace. To assert that such a hook has been added, you can use the special `closure` type:
+
+```php
+public function test_anonymous_function_hook() {
+	\WP_Mock::expectActionAdded( 'save_post', \WP_Mock\Functions::type( 'closure' ) );
+	\WP_Mock::expectFilterAdded( 'the_content', \WP_Mock\Functions::type( 'closure' ) );
```

Comment:
Should these use `type( 'callable' )` instead?

```suggestion
	\WP_Mock::expectActionAdded( 'save_post', \WP_Mock\Functions::type( 'callable' ) );
	\WP_Mock::expectFilterAdded( 'the_content', \WP_Mock\Functions::type( 'callable' ) );
```

Reply:
Or are we trying to match instances of the `Closure` class? In that case I think using `Clousure::class` is more accurate:

```suggestion
	\WP_Mock::expectActionAdded( 'save_post', \WP_Mock\Functions::type( Closure::class ) );
	\WP_Mock::expectFilterAdded( 'the_content', \WP_Mock\Functions::type( Closure::class ) );
```

`\WP_Mock\Functions::type( 'closure' )` works because `is_class( 'closure' )` returns `true` as class names are not case sensitive, but it took me a lot of digging to understand that.

Initially I thought there was some special handling for the `closure` string, but we are [passing that string to `Mockery::type();`](https://github.com/10up/wp_mock/blob/d0584171d602e60d1d58e2e0dccd35dcce529f18/php/WP_Mock/Functions.php#L99-L112) and [it is used as a class name](https://github.com/mockery/mockery/blob/3c3ca93df1752eb33ea9cfa9481c974e650b604c/library/Mockery/Matcher/Type.php#L40-L43).

If we are targeting the class name, I believe using `Closure::class` better reflects what's happening under the hood and makes the example more readable.

Reply:
@wvega-godaddy I have applied some of your suggestions in `feature/closures-bis`

Reply:
@wvega-godaddy see https://github.com/10up/wp_mock/pull/183